app/controllers/products_controller.rb

```diff
@@ -47,7 +47,7 @@ def destroy
 
   def search
     @query = params[:q]
-    @results = Product.where('name LIKE ?', "%#{@query}%")
+    @results = Product.where('name or size Like  ?', "%#{@query}%")
```

Comment:
@query variable used in only one place. you can directly use params[:q] in you query